source/PlayerInfo.cpp

```diff
@@ -741,14 +741,23 @@ Ship *PlayerInfo::Flagship()
 const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 {
 	if(!flagship)
-	{
 		for(const shared_ptr<Ship> &it : ships)
 			if(!it->IsParked() && it->GetSystem() == system && it->CanBeFlagship())
 			{
-				flagship = it;
-				break;
+				if(planet)
+				{
+					if(planet->CanLand(*it))
```

Comment:
Not critical, but I think this `if` and the `else` below can be avoided by using `if(!planet || planet->CanLand(*it))` on line 747 (new numbering). And then it is even possible to combine the conditions it with the expression above on line 754 (new numbering).